lib/client.js

```diff
@@ -183,11 +182,24 @@ class Client extends EventEmitter {
           })
         } else {
           this.socket.write('transfer-encoding: chunked\r\n', 'ascii')
-          var through = syncthrough(addTransferEncoding)
-          body.pipe(through)
-          through.pipe(this.socket, { end: false })
+
+          const onData = (chunk) => {
+            this.socket.write('\r\n' + Buffer.byteLength(chunk).toString(16) + '\r\n')
+            if (!this.socket.write(chunk)) {
+              body.pause()
+            }
+          }
+          const onDrain = () => {
+            body.resume()
+          }
+
+          body.on('data', onData)
+          this.socket.on('drain', onDrain)
+
           this.socket.uncork()
           eos(body, (err) => {
+            body.destroy(err)
```

Comment:
Should probably also destroy the body here.

Reply:
body is already finished here, so this is superfluous I think.
However, if the socket errors we should destroy the body. This is not done here.

Reply:
makes sense, I'll remove it.